From b0fb7d90d1a9ef8f766a8468413fc749791c0e0a Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 6 May 2015 15:11:34 +0100
Subject: [PATCH] p2v: Switch from matchbox-window-manager to metacity.

Matchbox only supports a single window on screen.  This worked fine
until we added the NetworkManager connection dialog -- which tries to
open in a new top level window, and of course fails.

Metacity supports multiple top-level windows so doesn't suffer this
problem.

Apart from that it's hard to tell the difference between the two.

(cherry picked from commit 9b2ca08ab543e2779a3a0da5f05b7ba4a5bc7b7d)
---
 p2v/launch-virt-p2v.in    |  2 +-
 p2v/p2v.ks.in             |  2 +-
 p2v/virt-p2v-make-disk.in | 10 +++++-----
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/p2v/launch-virt-p2v.in b/p2v/launch-virt-p2v.in
index a70e5c2..d2bafe3 100755
--- a/p2v/launch-virt-p2v.in
+++ b/p2v/launch-virt-p2v.in
@@ -22,7 +22,7 @@
 # and virt-p2v.
 if [ "$1" = "run" ]; then
     cd /
-    matchbox-window-manager &
+    metacity &
     exec @libexecdir@/virt-p2v
 else
     xinit "$0" run
diff --git a/p2v/p2v.ks.in b/p2v/p2v.ks.in
index 26bb925..6a3b24b 100644
--- a/p2v/p2v.ks.in
+++ b/p2v/p2v.ks.in
@@ -65,7 +65,7 @@ firewalld
 /usr/bin/Xorg
 xorg-x11-drivers
 xorg-x11-fonts-Type1
-matchbox-window-manager
+metacity
 pcre
 libxml2
 gtk2
diff --git a/p2v/virt-p2v-make-disk.in b/p2v/virt-p2v-make-disk.in
index fb02856..2bb364b 100644
--- a/p2v/virt-p2v-make-disk.in
+++ b/p2v/virt-p2v-make-disk.in
@@ -94,12 +94,12 @@ trap cleanup INT QUIT TERM EXIT ERR
 #   - Xorg drivers
 #   - some fonts
 #   - hardware support (firmware etc, RHBZ#1157679)
-#   - matchbox (window manager, another could be used)
+#   - metacity (window manager, another could be used)
 #
 # Note that libguestfs is NOT a dependency.
 case "$osversion" in
     centos-*|fedora-*|rhel-*|scientificlinux-*)
-        deps=pcre,libxml2,gtk2,/usr/bin/xinit,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/Xorg,xorg-x11-drivers,xorg-x11-fonts-Type1,matchbox-window-manager,@hardware-support
+        deps=pcre,libxml2,gtk2,/usr/bin/xinit,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/Xorg,xorg-x11-drivers,xorg-x11-fonts-Type1,metacity,@hardware-support
         cat > $tmpdir/p2v.conf <<'EOF'
 add_drivers+=" usb-storage "
 EOF
@@ -117,13 +117,13 @@ EOF
         "
         ;;
     debian-*|ubuntu-*)
-        deps=libpcre3,libxml2,libgtk2.0-0,openssh-client,qemu-utils,xorg,xserver-xorg-video-all,matchbox-window-manager
+        deps=libpcre3,libxml2,libgtk2.0-0,openssh-client,qemu-utils,xorg,xserver-xorg-video-all,metacity
         ;;
     archlinux-*)
-        deps=pcre,libxml2,gtk2,openssh,qemu,xorg-xinit,xorg-server,xf86-video-*,matchbox-window-manager
+        deps=pcre,libxml2,gtk2,openssh,qemu,xorg-xinit,xorg-server,xf86-video-*,metacity
         ;;
     opensuse-*|suse-*)
-        deps=pcre,libxml2,gtk2,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/xinit,/usr/bin/Xorg,xf86-video-*,matchbox-window-manager
+        deps=pcre,libxml2,gtk2,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/xinit,/usr/bin/Xorg,xf86-video-*,metacity
         ;;
     *)
         echo "$program: internal error: could not work out the Linux distro from '$osversion'"
-- 
1.8.3.1

